Added An Example With Tmux For Keyboard control

sandi-go opened the pull request:
package main

// import (
// 	"time"

// 	"gobot.io/x/gobot"
// 	"gobot.io/x/gobot/platforms/dji/tello"
// )

// func main() {
// 	drone := tello.NewDriver("8888")

// 	work := func() {
// 		drone.TakeOff()
// 		// var count int
// 		// gobot.Every(1*time.Second, func() {
// 		// drone.Hover()
// 		// if count == 30{
// 		// drone.Land()
// 		// }
// 		// drone.Clockwise(30)
// 		// })
// 		gobot.After(3*time.Second, func() {
// 			drone.BackFlip()
// 		})
// 		drone.BackFlip()

// 	}
// 	robot := gobot.NewRobot("tello",
// 		[]gobot.Connection{},
// 		[]gobot.Device{drone},
// 		work,
// 	)

// 	robot.Start()
// }

import (
	"fmt"
	"os"

	term "github.com/nsf/termbox-go"
	"gobot.io/x/gobot"
	"gobot.io/x/gobot/platforms/dji/tello"
	// "gobot.io/x/gobot/platforms/opencv"
	// "gocv.io/x/gocv"
)

const (
	frameSize = 960 * 720 * 3
)

func reset() {
	term.Sync() // cosmestic purpose
}
func main() {
	err := term.Init()
	if err != nil {
		panic(err)
	}
	defer term.Close()
	drone := tello.NewDriver("8890")
	work := func() {
		fmt.Println("Enter any key to see their ASCII code or press ESC button to quit")
		fmt.Println("Connected")
		for {
			switch ev := term.PollEvent(); ev.Type {
			case term.EventKey:
				switch ev.Key {
				case term.KeyEsc:
					drone.Land()
					os.Exit(9)
				case term.KeyF1:
					drone.BackFlip()
					reset()
					fmt.Println("F1 pressed")
				// case term.KeyF2:
				// 	reset()
				// 	fmt.Println("F2 pressed")
				// case term.KeyF3:
				// 	reset()
				// 	fmt.Println("F3 pressed")
				// case term.KeyF4:
				// 	reset()
				// 	fmt.Println("F4 pressed")
				// case term.KeyF5:
				// 	reset()
				// 	fmt.Println("F5 pressed")
				// case term.KeyF6:
				// 	reset()
				// 	fmt.Println("F6 pressed")
				// case term.KeyF7:
				// 	reset()
				// 	fmt.Println("F7 pressed")
				// case term.KeyF8:
				// 	reset()
				// 	fmt.Println("F8 pressed")
				// case term.KeyF9:
				// 	reset()
				// 	fmt.Println("F9 pressed")
				// case term.KeyF10:
				// 	reset()
				// 	fmt.Println("F10 pressed")
				// case term.KeyF11:
				// 	reset()
				// 	fmt.Println("F11 pressed")
				// case term.KeyF12:
				// 	reset()
				// 	fmt.Println("F12 pressed")
				// case term.KeyInsert:
				// 	reset()
				// 	fmt.Println("Insert pressed")
				// case term.KeyDelete:
				// 	reset()
				// 	fmt.Println("Delete pressed")
				// case term.KeyHome:
				// 	reset()
				// 	fmt.Println("Home pressed")
				// case term.KeyEnd:
				// 	reset()
				// 	fmt.Println("End pressed")
				case term.KeyPgup:
					drone.Up(9)
					reset()
					fmt.Println("Page Up pressed")
				case term.KeyPgdn:
					drone.Down(9)
					reset()
					fmt.Println("Page Down pressed")
				case term.KeyArrowUp:
					drone.Forward(9)
					reset()
					fmt.Println("Arrow Up pressed")
				case term.KeyArrowDown:
					drone.Backward(9)
					reset()
					fmt.Println("Arrow Down pressed")
				case term.KeyArrowLeft:
					drone.Left(9)
					reset()
					fmt.Println("Arrow Left pressed")
				case term.KeyArrowRight:
					drone.Right(9)
					reset()
					fmt.Println("Arrow Right pressed")
				case term.KeySpace:
					// drone.Hover()
					drone.BackFlip()
					reset()
					fmt.Println("Space pressed")
				case term.KeyBackspace:
					drone.Land()
					reset()
					fmt.Println("Backspace pressed")
					os.Exit(18)
				case term.KeyEnter:
					drone.TakeOff()
					reset()
					fmt.Println("Enter pressed")
				// case term.KeyTab:
				// 	reset()
				// 	fmt.Println("Tab pressed")

				default:
					// we only want to read a single character or one key pressed event
					reset()
					fmt.Println("ASCII : ", ev.Ch)

				}
			case term.EventError:
				panic(ev.Err)
			}
		}
		// drone.On(tello.ConnectedEvent, func(data interface{}) {
		// 	err := term.Init()
		// 	if err != nil {
		// 		panic(err)
		// 	}

		// })
	}

	robot := gobot.NewRobot("tello",
		[]gobot.Connection{},
		[]gobot.Device{drone},
		work,
	)

	robot.Start()
}

conejoninja committed:
Added RPi4 to documentation

Commit:
feat(ultrasonic): Add ultrasonic i2c

add ultrasonic support for i2c driver
add basic test for ultrasonic sensor
fix length and update test for i2c driver

tdaira committed:
Add delay for Run function of StepperDriver.

vladimir-kotikov committed:
Add method to publish MQTT messages with retain flag

danacr committed:
Updated .travis.yml to fix ci

danacr committed:
Temporary gocov patch for go.bug.st/serial.v1 cert

danacr committed:
Added tello edu driver

deadprogram commented:
Hello @sharmasandeepkr sorry for the delay in reviewing this PR. 

I do not see any commit in this pull request that contains a new example. Was that what you meant with the code in your comment?

danacr committed:
Removed temporary fix for the go.bug.st cert

tux-eithel committed:
#700 - Avoid to close the connection.

Each spi drivers must not close the connection: it's not its responsibility.
Robots should close the connection.

Signed-off-by: Eithel <tux.eithel@gmail.com>

mike1808 committed:
Add configuration for Nvidia Shield Controller

srydai4hmt committed:
Add HD44780 support

mike1808 committed:
Update ffmpeg command to decrease latency

* Add options to remove cache and make video processing in real-time
* Add explanation to how download weights

hnw committed:
Initial support for SHT2x Temperature/Humidity sensor driver

Signed-off-by: Yoshio HANAWA <y@hnw.jp>

mike1808 committed:
Add graceful halt for Tello driver

* graceful terminate connections for video and command communication

stuartleeks committed:
Update function signatures in gpioTestAdaptor

stuartleeks committed:
Add tests for DigitalWrite value

stuartleeks committed:
Update RelayDriver to invert value written on Inverted

trevrosen committed:
Merge pull request #742 from stuartleeks/sl/relay-inverted

Update value written via DigitalWrite if RelayDriver.Inverted is true

bradshjg committed:
Typo in PublishWithQOS godoc

* `poken.Token` -> `paho.Token`

Commit:
Use go.mod friendly version of go.bug.st/serial library

imle committed:
Resolve issue with PWM for PWMWrite

4096 is outside of the value that is able to be written to a pin. Lowered the scale and updated the comments

jedipunkz committed:
add symbol keys for platform/keyboard

Commit:
Support for BMP388 Barometric Pressure/Temperature/Altitude Sensor

Commit:
[test] Implement driver tests

deadprogram committed:
ble: initial take on working go-bluetooth replacement for go-ble

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
ble: correctly able to convert the UUID format to that needed by go-bluetooth

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
modules: update dependencies to pickup feature branch of go-bluetooth with latest macOS changes

Signed-off-by: Ron Evans <ron@hybridgroup.com>

deadprogram committed:
ble: updates so macOS works

Signed-off-by: Ron Evans <ron@hybridgroup.com>

deadprogram committed:
ble: scanresult channel needs to handle linux scan results

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
firmata: exclude ble adaptor from build on Windows

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
build: updates to appveyor.yml based on gocv file

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
modules: update to latest dev of go bluetooth package

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
build: add mingw-w64 to path

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
firmata: exclude ble adaptor from build on Windows

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
modules: update dependencies

deadprogram committed:
build: add circleci base

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
docs: remove Travis build badge

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
docs: add CircleCI build badge

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
build: remove remaining travisci related files

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
gpio: correct test for hd44780 driver to match newer signature

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
release: prepare for release 1.15.0

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
build: add codecov to CircleCI tests

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
build: separate out appveyor tests

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
docs: add badge for codecov

Signed-off-by: deadprogram <ron@hybridgroup.com>

ccary64 committed:
update uuid package and directly access it; remove archived uuid package

bkmeneguello committed:
Guards Dji Tello Halt against nil dereference

Fixes #778

yurigorokhov committed:
Adding support for hmc8553l compass

gandaldf committed:
Fix servo and DC motors presence

Fix the presumption both servo and DC motors are present

Commit:
digispark fix ReadByte & WriteByte, rework and add i2c tests

Commit:
digispark remove useless code in i2c test

Commit:
fix helper_test ReadByteData, ReadWordData to use reg

Commit:
drv2605l fix missing address write byte in test of Halt()

Commit:
bmp388 fix missing address write byte in test of Measurements

gen2thomas committed:
fix tinkerboard i2c0 to i2c4, improve comments in pin map, improve README

tfroidcoeur committed:
MCP3004: correct number of channels

the MCP3004 has 4 channels, not 3.

Fixes: b0160105("spi: Add MCP3202, MCP3204, MCP3208, and MCP3304
drivers")

Signed-off-by: Tim Froidcoeur <tim.froidcoeur@gmail.com>

gen2thomas committed:
start by rollback code to state before 2018/06/21

gen2thomas committed:
FIX the bug #568 without further impact, heavy improvements of tests

gen2thomas committed:
prepare refactoring of PR569

gen2thomas committed:
failing leftovers after usage of PR #569

gen2thomas committed:
ReadGPIO fixed with #576, failing leftovers for PinMode, SetPullUp and SetPolarity

gen2thomas committed:
fixed PinMode, SetPullUp and SetPolarity, unit tests activated

gen2thomas committed:
final cleanup of code and tests

gen2thomas committed:
introduce speed optimizations and related tests

gen2thomas committed:
add tests, shrink access to some internal variables

gen2thomas committed:
switch off debug

gen2thomas committed:
introduce adafruit1109 2x16 LCD with 5 keys

gen2thomas committed:
add AnalogActuatorDriver, analog temperature sensor, driver for PCF8591 (with 400kbit stabilization), driver for YL-40

gen2thomas committed:
don't panic on 'With*' allow simpler wrapping of drivers

gen2thomas committed:
mcp23017: add mutex for write, hd44780: fix mutexes

DiscreteTom committed:
fix raspi PWMPin.SetDutyCycle (#800)

raspi: PWMPin.SetDutyCycle
   - value 0 is allowed in pi blaster
   - for pi-blaster, duty should not going below 2us, unless the duty is 0

deadprogram committed:
all: prepare for release v1.16.0

Signed-off-by: deadprogram <ron@hybridgroup.com>

gen2thomas commented:
@sharmasandeepkr and @deadprogram because this PR has no answer to the last question since more than 2 years, I'm going to close the PR without any merge.
